### PR TITLE
📝 Prepare for Dedicated Gen 3 release

### PR DESCRIPTION
**Pull request by CollierCZ**
## Why

Dedicated Gen 3 will be released soon. The names of things are changing as part of that.
But may change again not too long after.

## What's changed

- ✨ Allowed shortcodes in front matter and note titles
- 🔥 Removed outdated upgrade instructions
- ✅ Added a new script to check internal links
  - Has specific exceptions (not just ignoring entire files)
  - Runs in pre-commit hook and in workflow
- 📝 Switch to variables for Dedicated generations
- 📝 Update RabbitMQ versions for Gen 3
- Upgraded Hugo to get shortcodes working everywhere
- Move files under `/dedicated` to `/dedicated-gen-2`

### CODE REVIEW

**Review by trolologuy — APPROVED** [review]
✅  Great change, it will make future changes much easier.

- [ ] Final built pages are working as expected.
- [x] The information is correct.
- [x] The information is comprehensive.
- [x] The information is concise.
- [x] The information is secure.
- [x] The information is clear.
- [x] If anything has been moved or removed, there are clear redirects to the new location or related content.
- [x] All links work, including to section headers.
- [x] The changes follow the content and markup style guides.

🐞 Result page shows the short code [in some cases](https://dg3-release-tcqol6y-652soceglkw4u.eu-3.platformsh.site/search.html?q=dedicated).

💡 Could be nice to do the same for grid to be fully consistent.

File: docs/src/dedicated-gen-2/overview/_index.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## The {{% names/dedicated-gen-2 %}} cluster
+
+The {{% names/dedicated-gen-2 %}} cluster is a three-Virtual Machine redundant configuration provisioned by Platform.sh for each customer.
+Every service is replicated across all three virtual machines in a failover configuration (as opposed to sharding),
+allowing a site to remain up even if one of the virtual machines is lost entirely.
+
+The build process for your application is identical for both the Development Environment and the {{% names/dedicated-gen-2 %}} cluster.
+However, because the virtual machines are provisioned by Platform.sh, not as a container,
+service configuration must be done by Platform.sh's Customer Success team.
+By and large the same flexibility is available but only via filing a support ticket.
```
[the remark this answers is not in the document]
Comment (CollierCZ) [reply docs/src/dedicated-gen-2/overview/_index.md:30]: I didn't go through any of the text beyond the names. The entire thing could be improved.

**Review by trolologuy — DISMISSED** [review]
✅  The link preview is now working as expected.

File: docs/src/development/regions.md [hunk 2]
```diff
@@ -46,14 +46,14 @@ A legacy region is a region running an older version of the Platform.sh orchestr
 It doesn't include all available features.
 See a [list of not included features](../projects/region-migration.md#why-migrate-between-regions).
 
-These regions will be upgraded in the future.
+These regions are upgraded in the future.
```
Comment (trolologuy) [inline docs/src/development/regions.md:49]: :question:
This one sounds weird. Maybe we can stick with "will be"?

**Review by trolologuy — DISMISSED** [review]
✅

**Review by trolologuy — APPROVED** [review]
✅

**Pull request merged**